packages/z2s/src/compiler.ts

```diff
@@ -15,12 +16,21 @@ import {
   type SimpleCondition,
 } from '../../zero-protocol/src/ast.ts';
 import {clientToServer, NameMapper} from '../../zero-schema/src/name-mapper.ts';
-import type {TableSchema} from '../../zero-schema/src/table-schema.ts';
+import type {
+  TableSchema,
+  ValueType,
+} from '../../zero-schema/src/table-schema.ts';
 import type {Format} from '../../zql/src/ivm/view.ts';
 import {sql} from './sql.ts';
 
 type Tables = Record<string, TableSchema>;
 
+const ZQL_RESULT_KEY = 'zql_result';
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function extractZqlResult(pgResult: any): JSONValue {
+  return JSON.parse(pgResult[0][ZQL_RESULT_KEY]);
```

Comment:
We need the BigintJson variant, correct?

Reply:
Yes plan to do this in a follow up.  I'm still uncertain if with z2s we should throw if a BigInt value is > MAX_SAFE_INTEGER